test/test_launcher.rb

```diff
@@ -97,32 +99,42 @@ def test_puma_stats_clustered
       c.workers 1
       c.clear_binds!
     end
-    launcher = launcher(conf)
-    Thread.new do
+    launcher = create_launcher(conf)
+
+    status = nil
+
+    th_stats = Thread.new do
       sleep Puma::Configuration::DEFAULTS[:worker_check_interval] + 1
-      status = Puma.stats_hash[:worker_status].first[:last_status]
-      Puma::Server::STAT_METHODS.each do |stat|
-        assert_includes status, stat
-      end
+      status = Puma.stats_hash[:worker_status]&.first[:last_status]
```

Comment:
I remember looking at this some weeks ago but it was hard to follow... It doesn't look like you changed any behaviour so I will merge this as it sounds like very good changes for the test suite.